wordpress-mobile/WordPress-Android
# Mobile stories block - part 2: (on  "done" editing clicked)

This PR builds on top of https://github.com/wordpress-mobile/WordPress-Android/pull/12939

This PR has the work necessary to update a Story block once it's been edited in the StoryComposer.

Related Jetpack PR https://github.com/Automattic/jetpack/pull/17222 
Related Gutenberg PR https://github.com/WordPress/gutenberg/pull/25771 
Related mobile Gutenberg PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/2663 


**To test:**

Follow the steps provided in #17222 to create a story and be able to re-open it. Just in case, a reminder for the **creation** of a Story is here:
1. create a new Story post (main screen in WPAndroid: tap the FAB, then new Story)
2. add some slides, added views, etc. (you can choose media from the picker or tap the camera fab icon in the media picker to kick off the camera preview mode in the Stories library)
3. once you're done with it, tap on the button at the top-right corner of the screen to publish
4. the pre-publishing bottom sheet appears. Enter some title for your post and hit publish.
5. wait for it to get saved and uploaded.

**EDITING**
1. go to published Posts lists
2. tap on the Story post you just created
3. wait for Gutenberg to load
4. see the Story block appears. Select it and then tap on the tool icon that appears on the block
5. the Story composer will appear, loaded with the Story you created
6. confirm you can switch from slide to slide, and the added views are draggable as if it was the first time.
7. make any changes you want (even remove slides, add more slides, add more views, etc.) Note: the edge case of removing all slides is not covered for now - the Composer will go away after the last one has been removed and no edits will have been saved.
8. tap on DONE
9. the FrameSaveService will get triggered in the background and the user will land back at Gutenberg
10. observe the save progress updates get reflected on the block in gutenberg, as well as upload progress.
11. once everyting's ready, the block gets updated
12. tap on UPDATE to save he Post online
13. open your site and confirm you can see the story with the slides as you edited them, on the web.

**Known issues:**
-  Note on upload progress: we're not calculating the full story progress but paint from 0 to 100% of the progress line for each of the contained Story media files (the save process does reflect the full story save progress though). So depending on the amount of slides (media files) you have,  this may or may not be noticeable given all other files are uploaded in parallel. You could see first a "slower" line filling in and then a line going back and forth from 0 to 100 per each media file (should amount around for the same amount anyway. In any case the perception of "progress" is still there.
- if your first slide is has a video background, we just show an image placeholder for the time being.
- errors are not being handled as of yet, but happy to get any reports if you run into one.

PR submission checklist:
